src/main/java/de/hpi/bpt/argos/api/entityMapping/EntityMappingEndpointImpl.java

```diff
@@ -0,0 +1,241 @@
+            throw halt;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEntityMapping(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long mappingId = getMappingId(request);
+        EventEntityMapping mapping = PersistenceAdapterImpl.getInstance().getEventEntityMapping(mappingId);
+        if (mapping == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given mapping was not found");
+        }
+
+        // delete corresponding mapping conditions
+        List<MappingCondition> conditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mappingId);
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(conditions.toArray(new MappingCondition[conditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not delete mapping conditions");
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(mapping)) {
+            halt(HttpStatusCodes.ERROR, "could not delete mapping");
+        }
+
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String editEntityMapping(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        JsonObject jsonBody = new JsonObject();
+        try {
+            jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+        } catch (JsonSyntaxException e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.BAD_REQUEST, "not a valid json");
+        }
+
+        JsonObject jsonMapping = jsonBody.get(JSON_ENTITY_MAPPING_ATTRIBUTE).getAsJsonObject();
+        if (jsonMapping == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "no event mapping given in body");
+        }
+
+        EventEntityMapping oldMapping = PersistenceAdapterImpl.getInstance().getEventEntityMapping(getMappingId(request));
+
+        // update conditions
+        List<MappingCondition> oldConditions = PersistenceAdapterImpl.getInstance().getMappingConditions(oldMapping.getId());
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(oldConditions.toArray(new MappingCondition[oldConditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not delete conditions");
+        }
+
+        JsonArray jsonConditions = jsonMapping.get(MAPPING_CONDITIONS_ATTRIBUTE).getAsJsonArray();

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'jsonMapping' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)